lib/automation/llm_triage.rb

```diff
@@ -82,6 +83,25 @@ def self.handle(
               force_review: true,
             )
           end
+
+          # note this does not notify user of the action
+          # as it is mainly used for spam where notification
+          # is not needed.
+          # we will need another flag if we want to notify
+          if hide_post
+            reason ||=
+              (
+                if post.hidden_at
+                  Post.hidden_reasons[:flag_threshold_reached_again]
+                else
+                  Post.hidden_reasons[:flag_threshold_reached]
+                end
+              )
+
+            post.update!(hidden: true, hidden_at: Time.zone.now, hidden_reason_id: reason)
+
+            post.topic.update!(visible: false) if post.post_number == 1
```

Comment:
Should use `topic.update_status` for this, example here:

https://github.com/discourse/discourse/blob/c446caa6e9f68be24bdc3d0ff0b33e0ddc2130c6/app/models/post.rb#L628-L633